stdlib/3/io.pyi

```diff
@@ -63,8 +63,10 @@ class BufferedIOBase(IOBase):
     def write(self, __buffer: Union[bytes, bytearray]) -> int: ...
     def readinto1(self, __buffer: _bytearray_like) -> int: ...
     def read(self, __size: Optional[int] = ...) -> bytes: ...
-    def read1(self, __size: int = ...) -> bytes: ...
-
+    if sys.version_info >= (3, 7):
```

Comment:
I think we should leave the BufferedIOBase definition alone; as far as I can tell, size should have a default value here in 3.5 and 3.6.
Instead, we should conditionally override in BytesIO itself (and BufferedReader and BufferedRandom, if you're up for it), type-ignoring if necessary.

Reply:
Hmm, this didn't seem to work out of the box, since now we conditionally override while changing the signature, causing things to fail in 3.5 and 3.6 once again:

```
stdlib/3/io.pyi:91: error: Signature of "read1" incompatible with supertype "BufferedIOBase"

stdlib/3/io.pyi:99: error: Signature of "read1" incompatible with supertype "BufferedIOBase"

stdlib/3/io.pyi:113: error: Signature of "read1" incompatible with supertype "BufferedIOBase"
```

What exactly do you mean by "type-ignoring if neccessary"? Maybe that's the solution, but I am unsure of what to do here.

Reply:
The solution is to add `# type: ignore` to the method definition. Sometimes real-world code violates the LSP so mypy will complain about it, but we should still reflect in the stub what the code actually does.

Reply:
Oh ok, it was way more obvious than I thought.